app/controllers/posts_controller.rb

```diff
@@ -20,14 +20,31 @@ def owed
     @hide_quicklinks = true
     @page_title = 'Replies Owed'
 
+    ids = PostAuthor.where(user_id: current_user.id)
+
     if params[:view] == 'hidden'
-      ids = PostAuthor.where(user_id: current_user.id, can_owe: false).group(:post_id).pluck(:post_id)
-      @posts = posts_from_relation(Post.where(id: ids).ordered) and return
+      ids = ids.where(can_owe: false)
+    else
+      ids = ids.where(can_owe: true)
+    end
+
+    ids = ids.group(:post_id).pluck(:post_id)
+    @posts = Post.where(id: ids)
+    unless params[:view] == 'hidden'
+      drafts = ReplyDraft.where(post_id: @posts.select(:id))
+      solo = PostAuthor.where(post_id: ids).group(:post_id).having('count(post_id) < 2').count.keys
```

Comment:
Can we do pluck(:post_id) here since we're not using the select for anything else?

Reply:
On the ReplyDraft one or? Not sure how to on the PostAuthor one